pkg/engine/ipaddresses.go

```diff
@@ -12,12 +12,15 @@ import (
 // When it's for master, this public ip address is created and added to the loadbalancer's frontendIPConfigurations
 // and it's created with the fqdn as name.
 // When it's for agent, this public ip address is created and added to the loadbalancer's frontendIPConfigurations.
-func CreatePublicIPAddress(isForMaster bool) PublicIPAddressARM {
+func CreatePublicIPAddress(isForMaster, includeDNS bool) PublicIPAddressARM {
```

Comment:
The new `includeDNS` param supports the private cluster + Standard LB scenario, which requires a public IP for master outbound access SNAT, but we don't attach a DNS record to it (that DNS record is for conveniently exposing inbound req's, e.g., apiserver, ssh to master)

Reply:
code nit:
- There are now two bools on the signature. Perhaps, the function could be decomposed.

Also, what happens in the function if someone calls `CreatePublicIPAddress(false, true)`? That seems like it would be subtly incorrect creating a public IP for the agent, but using the master dns settings. 

```go
func GetMasterDNSSettings() *network.PublicIPAddressDNSSettings { ... }
func CreateMasterPublicIPAddress() PublicIPAddressARM { ... }
func CreateAgentPublicIPAddress() PublicIPAddressARM { ... }
func createPublicIPAddress(name string) { ... }
```
or
```go
func SetMasterDNS(ip PublicIPAddressARM) PublicIPAddressARM { ... }
func CreatePublicIPAddress(name string) PublicIPAddressARM { ... }
```

Reply:
Agree this should be refactored. I think the addition of a 2nd bool to this func was a lighter footprint than decomposing this func, or creating a distinct "create public IP with no DNS" func.

Also agree that this invites a pathology, I think this approach is the least amount of code updates overall.

Would love to see some targeted, iterative refactoring in this surface area generally, these interfaces are pretty statically implemented at present. :/

Reply:
Sounds good. Let's defer and refactor when not pressed for time.